clientv3/balancer.go

```diff
@@ -354,3 +369,66 @@ func getHost(ep string) string {
 	}
 	return url.Host
 }
+
+func (b *simpleBalancer) trySwitchEndpoint() (doSwitch bool) {
+	var prevPinAddr string
+	var prevEp string
+	doSwitch = true
+
+	func() {
+		b.mu.Lock()
+		defer b.mu.Unlock()
```

Comment:
deferring will deadlock
1. acquire b.mu.Lock
2. block on b.updateAddrsC <- struct{}{}
3. updateNotifyLoop tries to acquire b.mu.RLock and deadlocks.

Reply:
OK, put it outside the func()